src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,261 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.Objects;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final int DEFAULT_LENGTH_FOR_LOGIN_AND_PASSWORD = 6;
+    private static final int EIGHTEEN_YEARS_OLD = 18;
+    private static final long ZERO = 0;
+    private StorageDaoImpl storageDao;
+    private RegistrationServiceImpl registrationService;
+    private User user1;
+    private User user2;
+    private User user3;
+    private User user4;
+    private User user5;
```

Comment:
```suggestion
    private User validUser;
```
one or two valid users fields should be enough
you can use the same user to test notOk scenarios - setLogin("a") - assertThrows, setLogin("ab") on the same default user - assertThrows etc
clean up your tests

Reply:
++